src/wp-includes/admin-bar.php

```diff
@@ -275,10 +274,11 @@ function wp_admin_bar_my_account_item( $wp_admin_bar ) {
 		$profile_url = false;
 	}
 
-	$avatar = get_avatar( $user_id, 26 );
+	$avatar       = get_avatar( $user_id, 26 );
+	$current_user = wp_get_current_user();
+
 	/* translators: %s: Current user's display name. */
 	$howdy = sprintf( __( 'Howdy, %s' ), '<span class="display-name">' . $current_user->display_name . '</span>' );
```

Comment:
Since the `$current_user` is only used once, I'll update as follows in commit:

```suggestion
/* translators: %s: Current user's display name. */
$howdy = sprintf( __( 'Howdy, %s' ), '<span class="display-name">' . wp_get_current_user()->display_name . '</span>' );
```